pallets/funding/src/lib.rs

```diff
@@ -1054,8 +1027,10 @@ pub mod pallet {
 			jwt: UntrustedToken,
 			project_id: ProjectId,
 		) -> DispatchResultWithPostInfo {
-			let issuer = T::InstitutionalOrigin::ensure_origin(origin, &jwt, T::VerifierPublicKey::get())?;
-			Self::do_english_auction(issuer, project_id)
+			let (account, _did, investor_type) =
+				T::InvestorOrigin::ensure_origin(origin, &jwt, T::VerifierPublicKey::get())?;
+			ensure!(investor_type == InvestorType::Institutional, Error::<T>::NotAllowed);
+			Self::do_english_auction(account, project_id)
```

Comment:
I'm not entirely convinced about the new `ensure_origin` syntax, especially with the repetitive use of ensure (`ensure_origin` + `ensure!`). I get that in the Evaluation extrinsics, where we need to check whether the origin is Institutional OR Professional OR Retail, the old syntax feels a bit clunky. However, I believe we might strike a balance. What if we incorporate the `investor_type` check within the `ensure_origin` function and simply pass the minimum required investor type to it? We could introduce a hypothetical investor type called `Any` for instances where we're open to any credentialed user. Alternatively, we could pass an array of `InvestorTypes` or even use numerical representation for Retail (0), Professional (1), and Institutional (2) to denote the minimum required "rank". In summary: I think there are several ways we could explore to refine the current syntax even further.